vignettes/data-combined.Rmd

```diff
@@ -0,0 +1,300 @@
+myDataCombined$addDataSets(
+  obsData$`Vergin 1995.Iv`, 
+  groups = "Aciclovir PVB"
+)
+```
+
+Every data, be it simulated results or from `DataSet`, must have a unique name 
+within `DataCombined`. If not specified by user, the path of simulated results 
+or the `$name` property of the `DataSet` are used as the name. Alternatively, we 
+can define the name when adding the data, as in the above example adding simulated 
+results.
+
+There are a few things to keep in mind here:
+
+- It doesn't matter in which order observed or simulated datasets are added.
+- You can add multiple `DataSet` objects in `$addDataSets()` method call.
+- You can add only a single instance of `SimulationResults` in `$addSimulationResults()` method call.
+- If you add a dataset with the same name as existing (e.g., adding simulation results 
+from multiple simulations and not specifying the name, so the path of the results 
+is used as the name), the new data will replace the existing.
+
+## Grouping
+
+Since the `DataCombined` object can store many datasets, some of them may naturally form a grouping and you would wish to track this in the object. Both `$addDataSets()` and `$addSimulationResults()` allow group specification via `groups` argument.
+
+When being plotted, data sets without a grouping will appear with distinct color, 
+line style (for simulated results), or symbol style (for observed data) with a 
+separate entry in the legend, with the name of the data set being the legend entry.
+
+Let's create a `DataCombined` and add simulation results and observed data *without* 
+specifying their grouping and create a time profile:
+
+```{r}
+myDataCombined <- DataCombined$new()
+
+myDataCombined$addSimulationResults(
+  simulationResults = simResults,
+  quantitiesOrPaths = outputPath,
+  names = "Aciclovir Plasma"
+)
+
+myDataCombined$addDataSets(
+  obsData$`Vergin 1995.Iv`
+)
+
+plotIndividualTimeProfile(dataCombined = myDataCombined)
+```
+
+If you do not specify `groups` when you add datasets or set it to `NULL`, and 
+wish to update groupings later, you can use the `$setGroups()` method. All data 
+within one group will get one entry in the legend with the name of the group, 
+and will be plotted with the same color.
+
+```{r}
+myDataCombined$setGroups(names = c("Aciclovir Plasma", obsData$`Vergin 1995.Iv`$name),
+                         groups = "Aciclovir PVB")
+
+plotIndividualTimeProfile(dataCombined = myDataCombined)
+```
+
+At any point, you can check the current names and groupings with the following active field:
+
+```{r}
+myDataCombined$groupMap
+```
+
+## Transformations
+
+Sometimes the raw data included in `DataCombined` needs to be transformed using specified
+offset and scale factor values.
+
+This is supported via `$setDataTransformations()` method, where you can specify the names of datasets, and offsets and scale factors. If these arguments are scalar (i.e., of length 1), then the same value will be applied to all specified names.
+
```

Comment:
It would be good to state here, in which units the offsets must be given: **internal** units or **display** units ?

Reply:
> This is supported via `$setDataTransformations()` method, where you can specify the names of datasets, and offsets and scale factors. If these arguments are scalar (i.e., of length 1), then the same value will be applied to all specified names. Scale factors are unitless values by which raw data values will 
be multiplied, offsets are added to the raw values without any conversion and must 
therefore be in the same units as the raw data.